core/services/workflows/engine.go

```diff
@@ -143,10 +143,10 @@ type Engine struct {
 	clock          clockwork.Clock
 	ratelimiter    *ratelimiter.RateLimiter
 	workflowLimits *syncerlimiter.Limits
-	meterReport    *MeteringReport
+	meterReports   map[string]*MeteringReport
```

Comment:
It might be better to use a mutex for safety and wrap some functionality like `Add`, `Delete`, and `Get` or `GetOrNil`.